src/Emulator/Peripherals/Peripherals/Mocks/HPSHostController.cs

```diff
@@ -334,27 +347,154 @@ private bool RXNECleared()
+            socket.Listen(0);
+            this.Log(LogLevel.Info, "Server started waiting for client to connect...");
+            this.Log(LogLevel.Info, "Starting on: {0}", path);
+
+            currentSlave.Read(0);
+            var connection = socket.Accept();
+            this.Log(LogLevel.Info, "Connection established");
             
-            while(!registerAccess() && i < 8)
+            var commandBuffer = new byte[(int)NumberOfBytes.MessageType];
+            var prevWrite = false;
+            var length = 0;
+
+            while(SocketConnected(connection)){
+                PollForRegisterBit(RegisterBitName.OA1EN);
+                try {
+                    PollForSocketData(connection, commandBuffer);
+                }
+                catch(Exception e)
+                {
+                    this.Log(LogLevel.Warning, "{0} Exception caught.", e);
+                    return;
+                }
+                switch((SocketI2CMessageType)commandBuffer[0])
+                {
+                    case SocketI2CMessageType.Stop:
+                        currentSlave.FinishTransmission();
+                        break;
+                    // Parse read command
+                    case SocketI2CMessageType.Read:
+                        prevWrite = false;
+                        length = GetLength(connection);
+                    
+                        byte [] dataToSend =  GetLongBytes(length, timeInterval);
+                        try 
+                        {   
+                            PollForSocketData(connection, dataToSend, true);
+                        }
+                        catch(Exception e)
+                        {
+                            this.Log(LogLevel.Warning, "{0} Exception caught.", e);
+                            return;
+                        }
+                        break;
+                    // Parse write command
+                    case SocketI2CMessageType.Write:
+                        // Seems that the slave needs a read operation
+                        // to flush out the bits written
+                        if (prevWrite) {
+                            currentSlave.Read(0);
+                            PollForRegisterBit(RegisterBitName.RXNE);
+                        }
+                        prevWrite = true;
+                        length = GetLength(connection);
+                        byte[] data = new byte[length];
+                        try 
+                        {   
+                            PollForSocketData(connection, data);
+                        }
+                        catch(Exception e)
+                        {
+                            this.Log(LogLevel.Warning, "{0} Exception caught.", e);
+                            return;
+                        }
+                        if(data.Length == 1 && data[0] == 0x86){
+                            PollForError(timeInterval);
+                        }
```

Comment:
What is the purpose of this `if`? 

Should `IssueCommand(data)` (which seems to be very similar to `ReadError` in case of `data == [0x86]`) be called after `PollForError`?